packages/js/product-editor/src/blocks/product-fields/attributes/edit.tsx

```diff
@@ -27,12 +29,28 @@ export function AttributesBlockEdit( {
 
 	const blockProps = useWooBlockProps( attributes );
 
+	const {
+		attributes: attributeList,
+		fetchAttributes,
+		handleChange,
+	} = useProductAttributes( {
+		allAttributes: entityAttributes,
+		onChange: setEntityAttributes,
+		productId,
+	} );
+
+	useEffect( () => {
+		if ( isInSelectedTab ) {
+			fetchAttributes();
+		}
+	}, [ entityAttributes, isInSelectedTab ] );
+
 	return (
 		<div { ...blockProps }>
 			<AttributesContainer
-				productId={ productId }
+				attributeList={ attributeList }
```

Comment:
I think we better get rid/fully delete of the `AttributeContainer` component and use the `AttributeControl` in the same way variation-options block does. In the end `AttributeContainer` is just an inner layer that is not reused anywhere else. And after all you are now calling `useProductAttributes` also outside of `AttributeContainer` so I think component does not make sense now.

Reply:
I didn't delete the `Attributes` component since it's still being used [on WC-Admin](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/products/fills/attributes-section/attributes-field.tsx#L5), but I added [this warning](https://github.com/woocommerce/woocommerce/pull/47361/commits/9a4c3b01f310b9f0c4a368f4280c743877aa6b43) to the component.